react-components/src/library/components/stem-finder.js

```diff
@@ -229,6 +211,28 @@ const StemFinder = Component({
       // TODO: informal learning?
     })
 
+    return query.join('')
+  },
+
+  search: function (incremental) {
+    let displayLimit = incremental ? this.state.displayLimit + DISPLAY_LIMIT_INCREMENT : DISPLAY_LIMIT_INCREMENT
+
+    // short circuit further incremental searches when all data has been downloaded
+    if (incremental && (this.state.lastSearchResultCount === 0)) {
+      this.setState({
+        displayLimit: displayLimit
+      })
+      return
+    }
+
+    let resources = incremental ? this.state.resources.slice(0) : []
+    let searchPage = incremental ? this.state.searchPage + 1 : 1
+
+    let keyword = jQuery.trim(this.state.searchInput)
+    if (keyword !== '') {
+      ga('send', 'event', 'Home Page Search', 'Search', keyword)
```

Comment:
I'm not really following this refactor. Is this method called even on the autoSearch? If so does it mean that we will be logging a GA event for each of these auto searches?
Perhaps we should be doing that, but if we are, I think we should check with Ethan to make sure he is aware.

Reply:
No this is only called when the user hits enter in the search form or when the page is loaded with a search term in the query.  The `AutoSuggest` component has a separate call to get the suggestions.